extensions/smallrye-jwt/deployment/src/test/java/io/quarkus/jwt/test/dev/SmallryeJwtProcessorDevModeTest.java

```diff
@@ -0,0 +1,180 @@
+                .get("/token")
+                .andReturn()
+                .body()
+                .asString();
+
+        devMode.modifySourceFile("GreetingResource.java", s -> s.replace("Hello from Quarkus", "Hello from JWT"));
+
+        // there is no need to get another token
+        RestAssured.given()
+                .header(new Header("Authorization", "Bearer " + token))
+                .get("/only-user")
+                .then().assertThat().statusCode(200)
+                .body(Matchers.containsString("Hello from JWT"));
+    }
+
+    @Test
+    public void shouldUseTheSameTokenEvenWhenTheUserChangesTheConfigWithKeyProps() {
+        String token = RestAssured.given()
+                .header(new Header("Accept", "text/plain"))
+                .get("/token")
+                .andReturn()
+                .body()
+                .asString();
+
+        // there is no need to get another token
+        RestAssured.given()
+                .header(new Header("Authorization", "Bearer " + token))
+                .get("/only-user")
+                .then().assertThat().statusCode(200)
+                .body(Matchers.containsString("Hello from Quarkus REST"));
+
+        try {
+            String privateKey = KeyUtils.readKeyContent("/privateKey.pem");
+            String publicKey = KeyUtils.readKeyContent("/publicKey.pem");
+            devMode.modifyResourceFile("application.properties", s -> """
+                    smallrye.jwt.sign.key=%s
+                    mp.jwt.verify.publickey=%s
+                    """.formatted(privateKey, publicKey));
+        } catch (IOException e) {
+            fail("Was not possible for reading keys from resource");
+        }
+
+        // should throw error because the private/public are invalid
+        String newToken = RestAssured.given()
+                .header(new Header("Accept", "text/plain"))
+                .get("/token")
+                .andReturn()
+                .body()
+                .asString();
+
+        // should return 200 because the keys are valid
+        RestAssured.given()
+                .header(new Header("Authorization", "Bearer " + newToken))
+                .get("/only-user")
+                .then().assertThat().statusCode(200);
+
+        devMode.modifyResourceFile("application.properties", s -> "");
+
+        // there is no need to get another token
+        // should work with old token
+        RestAssured.given()
+                .header(new Header("Authorization", "Bearer " + token))
+                .get("/only-user")
+                .then().assertThat().statusCode(200)
+                .body(Matchers.containsString("Hello from Quarkus REST"));
+    }
+
+    @Path("/token")
+    static class TokenResource {
+
+        @GET
+        @Produces(MediaType.TEXT_PLAIN)
+        @PermitAll
+        public String hello() {
+            return Jwt.upn("jdoe@quarkus.io")
+                    .issuer("https://quarkus.io/issuer")
```

Comment:
@mcruzdev Sorry, the fact that the server does not have to set this issuer property in this case is good, but you can have a smallrye-jwt-build property set to have issuer configured, please do it ad then `.issuer(...)` can be dropped